joblib/numpy_pickle_utils.py

```diff
@@ -150,9 +150,17 @@ def _read_fileobject(fileobj, filename, mmap_mode=None):
 
     """
     # Detect if the fileobj contains compressed data.
-    compressor = _detect_compressor(fileobj)
     if isinstance(fileobj, tuple(_COMPRESSOR_CLASSES)):
         compressor = fileobj.__class__.__name__
+    else:
+        # the fileobj is not part of the supported compressor classes
+        # let's try to determine the compressor by reading the magic number at
+        # the very beginning of the file.
+        if PY3_OR_LATER and not fileobj.seekable():
+            # With Python 3 file object have to be able to seek explicitely.
+            raise ValueError("Input fileobject must be seekable.")
+        compressor = _detect_compressor(fileobj)
```

Comment:
I am +1 for moving the _detect_compressor here, but I am not sure what the added value is of requiring that `.seekable()` return True.

What kind of error do we get with a file that is not seekable?